packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts

```diff
@@ -0,0 +1,538 @@
+    revealTicketTimestampConsumed: fieldsToReveal.revealTimestampConsumed
+      ? "1"
+      : "0",
+    ticketTimestampSigned: ticketAsBigIntArray[4].toString(),
+    revealTicketTimestampSigned: fieldsToReveal.revealTimestampSigned
+      ? "1"
+      : "0",
+    ticketAttendeeSemaphoreId: ticketAsBigIntArray[5].toString(),
+    revealTicketAttendeeSemaphoreId: fieldsToReveal.revealAttendeeSemaphoreId
+      ? "1"
+      : "0",
+    ticketIsConsumed: ticketAsBigIntArray[6].toString(),
+    revealTicketIsConsumed: fieldsToReveal.revealIsConsumed ? "1" : "0",
+    ticketIsRevoked: ticketAsBigIntArray[7].toString(),
+    revealTicketIsRevoked: fieldsToReveal.revealIsRevoked ? "1" : "0",
+
+    // Tickjet signature fields
+    ticketSignerPubkeyAx: babyJub.F.toObject(
+      fromHexString(pubKey[0])
+    ).toString(),
+    ticketSignerPubkeyAy: babyJub.F.toObject(
+      fromHexString(pubKey[1])
+    ).toString(),
+    ticketSignatureR8x: babyJub.F.toObject(rawSig.R8[0]).toString(),
+    ticketSignatureR8y: babyJub.F.toObject(rawSig.R8[1]).toString(),
+    ticketSignatureS: rawSig.S.toString(),
+
+    // Attendee identity secret
+    semaphoreIdentityNullifier: identityPCD.claim.identity
+      .getNullifier()
+      .toString(),
+    semaphoreIdentityTrapdoor: identityPCD.claim.identity
+      .getTrapdoor()
+      .toString(),
+
+    // Valid event ID list
+    validEventIds: snarkInputForValidEventIds(validEventIdsInput),
+    checkValidEventIds: checkValidEventIds ? "1" : "0",
+
+    // Security features
+    externalNullifier:
+      externalNullifer || STATIC_TICKET_PCD_NULLIFIER.toString(),
+    revealNullifierHash: externalNullifer ? "1" : "0",
+    watermark: watermark.toString()
+  };
+}
+
+function claimFromProofResult(
+  ticketPCD: EdDSATicketPCD,
+  publicSignals: string[],
+  validEventIds: string[] | undefined,
+  externalNullifer: string | undefined,
+  watermark: bigint
+): ZKEdDSAEventTicketPCDClaim {
+  const partialTicket: Partial<ITicketData> = {};
+  if (!babyJubIsNegativeOne(publicSignals[0])) {
+    partialTicket.ticketId = decStringToBigIntToUuid(publicSignals[0]);
+  }
+  if (!babyJubIsNegativeOne(publicSignals[1])) {
+    partialTicket.eventId = decStringToBigIntToUuid(publicSignals[1]);
+  }
+  if (!babyJubIsNegativeOne(publicSignals[2])) {
+    partialTicket.productId = decStringToBigIntToUuid(publicSignals[2]);
+  }
+  if (!babyJubIsNegativeOne(publicSignals[3])) {
+    partialTicket.timestampConsumed = parseInt(publicSignals[3]);
+  }
+  if (!babyJubIsNegativeOne(publicSignals[4])) {
+    partialTicket.timestampSigned = parseInt(publicSignals[4]);
+  }
+  if (!babyJubIsNegativeOne(publicSignals[5])) {
+    partialTicket.attendeeSemaphoreId = publicSignals[5];
+  }
+  if (!babyJubIsNegativeOne(publicSignals[6])) {
+    partialTicket.isConsumed = publicSignals[6] !== "0";
+  }
+  if (!babyJubIsNegativeOne(publicSignals[7])) {
+    partialTicket.isRevoked = publicSignals[7] !== "0";
+  }
```

Comment:
there are so many of these manual repetitive operations in this PCD that it's hard for me to be sure you did it right.

Reply:
in dark forest we did something like [this](https://github.com/darkforest-eth/client/blob/master/src/_types/darkforest/api/ContractsAPITypes.ts) to get something that felt a bit better than the "array of signals" thing